backend/src/akvo/lumen/lib/auth.clj

```diff
@@ -0,0 +1,77 @@
+(ns akvo.lumen.lib.auth
+  (:require
+   [akvo.commons.jwt :as jwt]
+   [akvo.lumen.component.flow :as c.flow]
+   [akvo.lumen.component.tenant-manager :as tenant-manager]
+   [akvo.lumen.protocols :as p]
+   [clojure.spec.alpha :as s]
+   [clojure.string :as str]
+   [clojure.tools.logging :as log]
+   [hugsql.core :as hugsql]
+   [integrant.core :as ig]
+   [reitit.core :as rc]))
+
+(defrecord AuthorisedDBQueryService [tenant-conn authorised-uuid-tree]
+  p/DBQueryService
+  (p/query [this fun param-data options command-options]
+    (log/info :AuthorisedDBQueryService :query fun :param-data param-data :options options :command-options command-options :authorised-uuid-tree authorised-uuid-tree)
+    (apply fun [(:tenant-conn this) (merge param-data authorised-uuid-tree)
+                options command-options]))
+  (p/query [this fun param-data options]
+    (p/query this fun param-data options nil))
+  (p/query [this fun param-data]
+    (p/query this fun param-data nil))
+  (p/query [this fun]
+    (p/query this fun nil)))
+
+(defn new-dbqs
+  "`authorised-uuid-tree`: {:auth-datasets [] :auth-visualisations [] :auth-dashboards [] :auth-collections []}"
+  [tenant-conn authorised-uuid-tree]
+  (AuthorisedDBQueryService. tenant-conn authorised-uuid-tree))
+
+(hugsql/def-db-fns "akvo/lumen/lib/dataset.sql")
+
+(defn match-by-jwt-family-name?
```

Comment:
If we have the energy, I think we should lift this out of this namespace and do something more holistic so we can reuse for other new features. Maybe even add another claim for "staff" or "bleeding-edge".. It feels like looking for given names will be short lived.

Reply:
resolving conversation after agreeing that we keep this family-name flag on this development level